### PR TITLE
DOC: Add dependency table in readme. Closes #394

### LINKED ISSUE #394: Dependency links in readme
Very few users will come to ODL and expect to use it without its core dependencies. We should therefore add a few words on the dependencies and links to their respective webpages/repositories to the README.md file.

### PR DESCRIPTION
**Pull request by adler-j**
* Add table with optional dependencies
* Change name of some pip install options
* Minor formatting of readme

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -135,9 +135,10 @@ def run_tests(self):
     extras_require={
         'testing': test_requires,
         'show': 'matplotlib',
-        'fft': 'pyfftw',
-        'wavelets': 'Pywavelets',
-        'all': ['Pywavelets', 'pyfftw', 'matplotlib']
+        'fftw': 'pyfftw',
+        'pywavelets': 'Pywavelets',
+        'scikit' : 'scikit-image',
+        'all': test_requires + ['Pywavelets', 'pyfftw', 'matplotlib']
```
Comment (kohr-h) [inline setup.py]: no scikit-image?

File: README.md [hunk 2]
```diff
@@ -37,7 +37,26 @@ You can check that everything was installed properly by running
...
+
+where FOO, BAR, etc are the dependencies. Others require more complicated installation procedures, in that case, consult the respective package for instructions.
+
+| Package | Purpose | Pip |
+|---------|---------|-----|
+| [ASTRA](https://github.com/astra-toolbox/astra-toolbox)    | Tomographic forward/backward projectors in [RayTransform](http://odl.readthedocs.io/generated/odl.tomo.operators.ray_trafo.RayTransform.html). |  |
+| [scikit-image](http://scikit-image.org/)    | 2D parallel beam forward/backward projectors in [RayTransform](http://odl.readthedocs.io/generated/odl.tomo.operators.ray_trafo.RayTransform.html). | scikit |
+| [FFTW](https://github.com/pyFFTW/pyFFTW)   | Accelerated [FourierTransform](http://odl.readthedocs.io/generated/odl.trafos.fourier.FourierTransform.html). | fftw |
+| [PyWavelets](https://github.com/PyWavelets/pywt)   | Computation of the  [WaveletTransform](http://odl.readthedocs.io/generated/odl.trafos.wavelet.WaveletTransform.html). | pywavelets |
+| [matplotlib](http://matplotlib.org/)   | Visualization through the [show](http://odl.readthedocs.io/generated/odl.discr.lp_discr.DiscreteLpVector.show.html) command. | show |
```
Comment (kohr-h) [inline README.md]: PIP column says "show" instead of "matplotlib"

File: README.md [hunk 3]
```diff
@@ -37,7 +37,26 @@ You can check that everything was installed properly by running
...
+| [py.test](http://pytest.org/latest/)   | Automatic tests for ODL. | testing |
```
Comment (kohr-h) [inline README.md]: The package is called `pytest`

File: README.md (hunk 2 above)
Comment (adler-j) [inline README.md]: It says show in the setup.py file. Which one do we change?

**Comment by kohr-h** [comment]
Done reviewing.

File: README.md (hunk 2 above)
Comment (kohr-h) [inline README.md]: Ah okay, then I misunderstood your table. I thought you were listing the Pip package names. Perhaps "Dependency" would be a better column name then?

Comment (adler-j) [inline README.md]: Good idea!

File: setup.py (hunk 1 above)
Comment (adler-j) [inline setup.py]: Good catch

**Pull request merged**